extensions/ads-windows-support/src/main.ts

```diff
@@ -0,0 +1,97 @@
+const localize = nls.loadMessageBundle();
+
+let exePath:string;
+
+export function activate(context: vscode.ExtensionContext): Promise<void> {
+
+    // Only supported on Win32 currently, display error message if not that until extensions are able to block install
+    // based on conditions
+    if(process.platform === 'win32') {
+        let config: IConfig = JSON.parse(JSON.stringify(baseConfig));
+        config.installDirectory = path.join(context.extensionPath, config.installDirectory);
+        config.proxy = utils.getConfiguration('http').get('proxy');
+        config.strictSSL = utils.getConfiguration('http').get('proxyStrictSSL') || true;
+
+        const serverdownloader = new ServerProvider(config);
+        const installationStart = Date.now();
+
+        return new Promise((resolve, reject) => {
+            serverdownloader.getOrDownloadServer().then(e => {
+                const installationComplete = Date.now();
+
+                Telemetry.sendTelemetryEvent('startup/ExtensionStarted', {
+                    installationTime: String(installationComplete - installationStart),
+                    beginningTimestamp: String(installationStart)
+                });
+                serverdownloader.getServerPath().then(path =>
+                    {
+                        // Don't register the command if we couldn't find the EXE since it won't be able to do anything
+                        if(path) {
+                            exePath = path;
+                            context.subscriptions.push(
+                                vscode.commands.registerCommand('adsWindowsSupport.launchSsmsDialog', handleLaunchSsmsDialogCommand));
+                        }
+                        resolve();
+                    });
+            }, e => {
+                Telemetry.sendTelemetryEvent('startup/ExtensionInitializationFailed');
+                // Just resolve to avoid unhandled promise. We show the error to the user.
+                resolve();
+            });
+        });
+    } else {
+        vscode.window.showErrorMessage(localize('adsWindowsSupport.onlySupportedOnWindows', 'The ADS Windows Support Extension is only supported on Windows platforms.'));
+    }
+}
+
+/**
+ * Handler for the commmand to launch an SSMS dialog using SsmsMin.exe
+ * @param params The params used to construct the command
+ */
+function handleLaunchSsmsDialogCommand(params: sqlops.adsWindowsSupport.LaunchSsmsDialogParams) {
+    Telemetry.sendTelemetryEvent('LaunchSsmsDialog', { 'action': params.action});
+    let args = buildSsmsMinCommandArgs(params);
+    var proc = shelljs.exec(
+        /*command*/`"${exePath}" ${args}`,
+        /*options*/'',
+        (code, stdout, stderr) => {
+            Telemetry.sendTelemetryEvent('LaunchSsmsDialogResult', {
+                'action': params.action,
+                'returnCode': code
+            });
+        });
+
+    // If we're not using AAD the tool prompts for a password on stdin
```

Comment:
how does this work?  the user enters a password in plaintext at a command window?